### PR TITLE
Licensing: Replace connection is_active

### PR DESCRIPTION
**Pull request by fgiannar**
Replace Connection's `is_active` check with `has_connected_owner`.
With user-less in mind we will be slowly deprecating the is_active method in the Manager class.

Changes proposed in this Pull Request:
`is_active` was audited and replaced by `has_connected_owner`, which indicates an established user-level connection.

Jetpack product discussion
1191179647901802-as-1200087466875449

Does this pull request change what data or activity we track or use?
No

Testing instructions:
Basically look at the code to validate the logic.

### CODE REVIEW

**Comment by leogermani** [comment]
Hi @atanas-dev , to give you a bit of context:

We will be slowing deprecating the `Connection_Manager::is_active` method. Packages should use a more low-level function to check the current connection status.

If a user connection is required, we should probably use `has_connected_owner`, but if the feature can work without a connected user, checking for `is_connected` is enough.

In the case of Licenses, we understand that, for now, we still require a user. So the change proposed in this PR does not modify the current behavior, because that's also what `is_active` is checking.

We just wanted to confirm that this looks good for you too.

**Review by atanas-dev — APPROVED** [review]
Thanks for the ping - we do need a user so the change LGTM!

**Review by leogermani — APPROVED** [review]
